app/controllers/users/sessions_controller.rb

```diff
@@ -55,7 +55,13 @@ def track_authentication_attempt(email)
 
     def cache_active_profile
       cacher = Pii::Cacher.new(current_user, user_session)
-      cacher.save(current_user.user_access_key)
+      begin
+        cacher.save(current_user.user_access_key)
+      rescue Pii::EncryptionError => _err
+        current_user.active_profile.deactivate
+        properties = { user_id: current_user.uuid }
+        analytics.track_event(Analytics::PROFILE_ENCRYPTION_INVALID, properties)
```

Comment:
the problem was that the encrypted_pii was not correctly encrypted, and so it could not be decrypted. tomato? tomahto?

I'd prefer not to have a proliferation of exception classes here. What matters imo is that something went irretrievably wrong and we had to give up somewhere in the encryption pipeline.

I've added error messages in addition to the `raise` so that it is more clear (I hope). Mostly this is flow-control-via-exceptions. The error messages are never exposed to the user.